lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -383,6 +433,811 @@ void commitToModel(GeneratedCodeBasicInfo &GCBI,
+    if (GCBI::isJumpTarget(&BB)) {
+      auto Start = getBasicBlockPC(&BB);
+      MetaAddress End;
+      uint64_t Size;
+
+      bool Stop = false;
+      model::BasicBlock Block{ Start };
+      OnceQueue<BasicBlock *> Queue;
+      Queue.insert(&BB);
+
+      while (!Queue.empty()) {
+        BasicBlock *Current = Queue.pop();
+
+        auto [CandidateEnd, CandidateSize] = getPC(Current->getTerminator());
+        if (CandidateEnd > End) {
+          End = CandidateEnd;
+          Size = CandidateSize;
+        }
+
+        for (auto *Succ : successors(Current)) {
+          if (GCBI::isJumpTarget(Succ)) {
+            auto Destination = getBasicBlockPC(Succ);
+            auto Edge = makeEdge(Destination,
+                                 model::FunctionEdgeType::DirectBranch);
+            Block.Successors.insert(Edge);
+          } else if (UnexpectedPCCloned == Succ && succ_size(Current) == 1) {
+            Block.Successors.insert(makeEdge(MetaAddress::invalid(),
+                                             model::FunctionEdgeType::LongJmp));
+          } else {
+            auto *I = Succ->getFirstNonPHI();
+
+            if (isa<CallInst>(I)
+                && cast<CallInst>(I)->getCalledFunction() == PreHookMarker) {
+              MetaAddress Destination;
+              model::FunctionEdgeType::Values Type;
+              auto *CalleePC = cast<CallInst>(I)->getArgOperand(1);
+
+              // Direct or indirect call?
+              if (isa<ConstantStruct>(CalleePC)) {
+                Destination = MetaAddress::fromConstant(CalleePC);
+                Type = model::FunctionEdgeType::FunctionCall;
+              } else {
+                Destination = MetaAddress::invalid();
+                Type = model::FunctionEdgeType::IndirectCall;
+              }
+
+              auto Edge = makeEdge(Destination, Type);
+              auto DestTy = Oracle.getFunctionType(Destination);
+              if (DestTy == FunctionTypeValue::NoReturn)
+                cast<model::CallEdge>(Edge.get())->IsNoReturn = true;
+
+              Block.Successors.insert(Edge);
+            } else if (auto *Call = getCallTo(I, "function_call")) {
+              // Marker can be erased now, no longer needed.
+              Call->eraseFromParent();
+
+              auto Destination = getBasicBlockPC(Succ);
+              auto Edge = makeEdge(Destination,
+                                   model::FunctionEdgeType::FakeFunctionCall);
+              Block.Successors.insert(Edge);
+            } else {
+              Queue.insert(Succ);
+            }
+          }
+        }
+      }
+
+      Block.End = End + Size;
+      CFG.insert(Block);
+    }
+  }
+
+  return CFG;
+}
+
+template<class FO>
+FunctionSummary
+CFEPAnalyzer<FO>::analyze(GlobalsArrayRef ABIRegisters,
+                          llvm::BasicBlock *Entry,
+                          llvm::raw_fd_ostream &OutputIBI,
+                          llvm::raw_fd_ostream &OutputAAWriter) {
```

Comment:
Whe  you have arguments of function methods that always have the same value, you should be consider to promote them to class fields.

Reply:
These values are initialized in `StackAnalysis::runOnFunction`, not in methods of a class. They are not class field but are passed to the method `analyze` because no other methods would use those values.

Reply:
Ah OK, I got it now. But does this make sense to do if they would be used only in one method? In this case, I'm passing this objects directly to `analyze` in each call, in the other case I'm constructing the object `CFEPAnalyzer` w/ those two fields every time in each call.

Reply:
`analyze` is the only public method.
As a user, I don't see why I need to pass it every time I call it instead of once in the constructor.
All the rest is an implementation detail that the user of your class shouldn't care about.

Reply:
OK, moved as class fields.

Reply:
Please review if this looks good.